### PR TITLE
Add `:trim` variable modifier

### LINKED ISSUE #1182: Add `:trim` variable modifier
**Is your feature request related to a problem? Please describe.**
The template tags allow variables definitions, but they include the white-spaces used for code organization.

```
    {if category_image}
        {layout:set name="header_image"}
            {category_image}
        {/layout:set}
    {/if}
```

**Describe the solution you'd like**
A `:trim` variable modifier could clean up variables on these cases.

```
{if layout:header_image:trim}
    <style>
    .main-header-and-search-form {
      background-image: url('{layout:header_image:trim}');
    }
    </style>
{/if}
```

**Describe alternatives you've considered**
The current solution is to code everyting in a single line.

```
    {if category_image}
        {layout:set name="header_image"}{category_image}{/layout:set}
    {/if}
```

**Teachability, Documentation, Adoption, Migration Strategy**
https://github.com/ExpressionEngine/ExpressionEngine-User-Guide/pull/201

- [x] I am capable and would like to work on implementation of this feature if it is considered.

### PR DESCRIPTION
**Pull request by robsonsobral**
## Overview

Add a variable modifier which returns a string with whitespace stripped from its beginning and end as defined at [PHP docs](https://www.php.net/manual/en/function.trim.php).

Resolves [#1182](https://github.com/ExpressionEngine/ExpressionEngine/issues/1182).

## Nature of This Change

- [ ] 🐛 Fixes a bug
- [x] 🚀 Implements a new feature
- [ ] 🛁 Refactors existing code
- [ ] 💅 Fixes coding style
- [ ] ✅ Adds tests
- [ ] 👽 Adds new dependency
- [ ] 🔥 Removes unused files / code
- [ ] 🔒 Improves security 

## Is this backwards compatible?

- [x] Yes
- [ ] No

## Documentation

User Guide Pull Request: https://github.com/ExpressionEngine/ExpressionEngine-User-Guide/pull/201

### CODE REVIEW

**Comment by intoeetive** [comment]
@robsonsobral would it be possible for you to add Cypress test for this addition?

Here's the example tests file from another PR (not merged yet) that covers some of existing modifiers, so you might use that as starting base: https://github.com/ExpressionEngine/ExpressionEngine/blob/feature/6.x/newlines-in-vars/tests/cypress/cypress/integration/field/modifiers.ee6.js

**Comment by robsonsobral** [comment]
Is it better, @intoeetive ? Again, I did it in te dark.

File: tests/cypress/cypress/integration/field/modifiers.ee6.js
Comment (intoeetive) [inline tests/cypress/cypress/integration/field/modifiers.ee6.js]: The URL seems to be wrong here, should be `modifiers/trim`

**Review by intoeetive — CHANGES_REQUESTED** [review]
Thank you for adding the tests. Unfortunately one of test cases is failing:

image

File: tests/cypress/cypress/integration/field/modifiers.ee6.js
Comment (robsonsobral) [reply tests/cypress/cypress/integration/field/modifiers.ee6.js]: SHAME ON ME! I'll not submit any commit without automatic tests, but I don't know how much time it will take!

Thank you for help and patience!

**Comment by robsonsobral** [comment]
Do you want me to rebase this branch, @intoeetive ?

**Comment by intoeetive** [comment]
Thank you - my local tests worked fine, now the automated tests on GitHub are running.
Once those finish, we will be good to schedule your PR to be included into one of coming EE releases (which might be not 6.1 though, as we almost locked that down)

JFYI, I have conflict in files because of your force push, but that was easy to resolve.

Thank you again for the contribution!

**Review by intoeetive — DISMISSED** [review]

**Comment by robsonsobral** [comment]
@intoeetive , don't you prefer I deal with the stale code on my PRs? You have a lot in your plate already, I think.

**Comment by intoeetive** [comment]
If you refer to "dismissed their stale review" message, that's done by GitHub automatically when code updates are made

**Comment by robsonsobral** [comment]
Or, no! I was thinking of update the code and deal with possible conflicts. Just that.

**Comment by intoeetive** [comment]
No worries. I was just making sure the latest updates with Cypress are caught up.
It is now passing all tests, so approving it to be included into next minor release.

Thank you for what you do. Everyone here at PacketTide thinks you do awesome things and help community a lot!

**Review by intoeetive — APPROVED** [review]

**Review by obfuscode — APPROVED** [review]

**Pull request merged**
